src/Actions.ts

```diff
@@ -79,13 +140,34 @@ export class Actions {
             return;
         }
 
+        // If the user just did /inorout we enter modal mode
+        if (req.body.text.trim().length === 0) {
+            try {
+                await this.displayModal(req.body.channel_id, req.body.trigger_id);
+                res.send();
+            } catch (err) {
+                // Better handling of when the bot isn't invited to the channel
+                if (err.data.error === "not_in_channel") {
+                    res.send("Bot must be invited to the channel before you can use it!");
+                } else {
+                    res.send(this.handleActionException(err).text);
+                }
+            }
+            return;
+        }
+
         // Create a new poll passing in the poll author and the other params
         const poll = Poll.slashCreate(`<@${req.body.user_id}>`, req.body.text.replace("@channel", "").replace("@everyone", "").replace("@here", "").split("\n"));
         try {
             await this.postMessage(req.body.channel_id, "A poll has been posted!", poll.getBlocks());
             res.send();
         } catch (err) {
-            res.send(this.handleActionException(err).text);
+            // Better handling of when the bot isn't invited to the channel
+            if (err.data.error === "not_in_channel") {
```

Comment:
Maybe also send the command text that a user entered to them as Slackbot or an inorout bot user so they can easily copy and paste afterwards?

Reply:
I'm going to consider this an enhancement and not in the scope of this PR. The only reason I added the `not_in_channel` error message was because I realized our current code just completely errors out and throws an error to the console and the user has no idea what's going on. This way slack will at least tell them they must invite the bot.